curvefs/src/client/rpcclient/mds_client.cpp

```diff
@@ -329,18 +333,24 @@ bool MdsClientImpl::GetMetaServerListInCopysets(
                 CopysetPeerInfo<MetaserverID> csinfo;
                 ::curvefs::mds::topology::MetaServerLocation csl =
                     info.cslocs(j);
-                uint16_t port = csl.port();
+                uint16_t internalPort = csl.port();
                 std::string internalIp = csl.hostip();
                 csinfo.peerID = csl.metaserverid();
+                uint16_t externalPort = internalPort;
                 std::string externalIp = internalIp;
+                if (csl.has_externalport()) {
+                    externalPort = csl.externalport();
+                }
                 if (csl.has_externalip()) {
                     externalIp = csl.externalip();
                 }
 
                 butil::EndPoint internal;
-                butil::str2endpoint(internalIp.c_str(), port, &internal);
+                butil::str2endpoint(internalIp.c_str(), internalPort,
+                    &internal);
                 butil::EndPoint external;
-                butil::str2endpoint(externalIp.c_str(), port, &external);
+                butil::str2endpoint(externalIp.c_str(), externalPort,
+                    &external);
```

Comment:
the logic is almost the same as above, try to extract this into a single function.

Reply:
> the logic is almost the same as above, try to extract this into a single function.

fixed